metrics.go

```diff
@@ -20,16 +20,91 @@ directly to Stackdriver Metrics.
 */
 
 import (
+	"errors"
 	"fmt"
+	"path"
 
 	"github.com/golang/protobuf/ptypes/timestamp"
 
 	distributionpb "google.golang.org/genproto/googleapis/api/distribution"
+	labelpb "google.golang.org/genproto/googleapis/api/label"
+	googlemetricpb "google.golang.org/genproto/googleapis/api/metric"
 	monitoringpb "google.golang.org/genproto/googleapis/monitoring/v3"
 
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 )
 
+var errNilMetric = errors.New("expecting a non-nil metric")
+
+func (se *statsExporter) protoToMonitoringMetricDescriptor(metric *metricspb.Metric) (*googlemetricpb.MetricDescriptor, error) {
```

Comment:
Should be *metricspb.MetricDescriptor -> *googlemetricpb.MetricDescriptor

Reply:
I thought about that but went against that signature because if we do that they we won't include Metric_Name https://godoc.org/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1#Metric_Name yet a Metric can support both for Descriptor https://godoc.org/github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1#Metric.Descriptor_

Reply:
<img width="751" alt="screen shot 2018-12-10 at 6 55 05 pm" src="https://user-images.githubusercontent.com/4898263/49775459-1fed7e80-fcad-11e8-8380-9a99ba16f161.png">

Reply:
and I also added tests to intentionally test for this

Reply:
Metric_Name is also from MetricDescriptor. It is kind of cached MetricDescriptor to save encoded bytes. See also https://github.com/census-instrumentation/opencensus-java/pull/1637.

Reply:
What happens though if we only get Metric_Name and not Metric_MetricDescriptor? We have to remember that this exporter will be fed all combinations of Metric

Reply:
That shouldn't happen - it's clients' responsibility to make sure they only sent Metric_Name for already sent metrics. 

Anyway this is an optimization, please file an issue and we can continue discussion there.

Reply:
I'll think about this and file and issue and we can talk more there as you've suggested.